R/verb-summarise.R

```diff
@@ -66,8 +113,21 @@ op_sort.op_summarise <- function(op) NULL
 #' @export
 sql_build.op_summarise <- function(op, con, ...) {
   select_vars <- translate_sql_(op$dots, con, window = FALSE, context = list(clause = "SELECT"))
-  group_vars <- c.sql(ident(op_grps(op$x)), con = con)
+  group_vars <- op_grps(op$x)
+  n <- length(group_vars)
+
+  .groups <- op$args$.groups
+  verbose <- summarise_verbose(.groups, caller_env())
+  # we cannot automatically infer "keep" as for local frames
+  # -> use `drop_last` as it was historically
+  .groups <- .groups %||% "drop_last"
+
+  if (verbose && identical(.groups, "drop_last") && n > 1) {
```

Comment:
I think you could move the `identical(.groups, "drop_last")` check into `summarise_verbose()`?